walk-in-interview/src/main/java/com/google/account/business/servlets/CreateBusinessServlet.java

```diff
@@ -0,0 +1,114 @@
+package com.google.account.business.servlets;
+
+import com.google.account.business.data.Business;
+import com.google.account.business.data.BusinessDatabase;
+import com.google.common.collect.ImmutableList;
+import com.google.firebase.auth.FirebaseAuthException;
+import com.google.utils.FirebaseAuthUtils;
+import com.google.utils.ServletUtils;
+import org.apache.commons.lang3.StringUtils;
+
+import javax.servlet.ServletException;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.util.Optional;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+import java.util.stream.Collectors;
+
+/** Servlet that handles creating new business account. */
+@WebServlet("/business-account")
+public final class CreateBusinessServlet extends HttpServlet {
+    private static final Logger LOGGER = Logger.getLogger(CreateBusinessServlet.class.getName());
+    private static final long TIMEOUT_SECONDS = 5;
+
+    private BusinessDatabase businessDatabase;
+
+    @Override
+    public void init() {
+        this.businessDatabase = new BusinessDatabase();
+    }
+
+    @Override
+    public void doPost(HttpServletRequest request, HttpServletResponse response) {
+        try {
+            Optional<String> optionalUid = FirebaseAuthUtils.getUid(request);
+
+            if (!optionalUid.isPresent()) {
+                LOGGER.log(Level.SEVERE, /* msg= */"Illegal uid");
+                response.setStatus(HttpServletResponse.SC_BAD_REQUEST);
+                return;
+            }
+
+            String uid = optionalUid.get();
+
+            // Gets business object from the client
+            Business inputBusiness = parseBusinessAccount(request);
+
+            // Fetches the existing business object from database
+            Optional<Business> existingBusinessOptional = this.businessDatabase.getBusinessAccount(uid).get();
+
+            Business business;
+
+            // Updates the preliminary business object at only the updatable fields
+            if (!existingBusinessOptional.isPresent()) {
+                business = Business.newBuilder()
+                        .setName(inputBusiness.getName())
+                        .setJobs(ImmutableList.of()).build();
+            } else {
+                business = Business.newBuilder()
+                        .setName(inputBusiness.getName())
```

Comment:
does that mean that only set fields will be updated? the others will remain untouched?

A better way might be:
```
business = existingBusinessOptional.toBuilder().setName(...).build();
```

note the "toBuilder" not "newBuilder".

OR even better better way is to NOT update the full business object but only the field.
Basically directly updating the name field in firebase instead of setting a name field to an empty business object and persisting the business object.